object_store/src/aws/precondition.rs

```diff
@@ -38,12 +39,18 @@ pub enum S3CopyIfNotExists {
     ///
     /// [`ObjectStore::copy_if_not_exists`]: crate::ObjectStore::copy_if_not_exists
     Header(String, String),
+    /// The same as `Header` but allows custom status code checking, for object stores that return values
+    /// other than 412.
+    HeaderWithStatus(String, String, reqwest::StatusCode),
 }
 
 impl std::fmt::Display for S3CopyIfNotExists {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         match self {
             Self::Header(k, v) => write!(f, "header: {}: {}", k, v),
+            Self::HeaderWithStatus(k, v, code) => {
+                write!(f, "header-with-status: {}: {k}: {v}", code.as_u16())
```

Comment:
```suggestion
                write!(f, "header-with-status: {k}: {v}: {}", code.as_u16())
```
I think the order should match the variant